app/core/Tracker2/SegmentTracker.js

```diff
@@ -103,49 +114,92 @@ export default class SegmentTracker extends BaseTracker {
   }
 
   async _initializeTracker () {
+    this.watchForDisableAllTrackingChanges(this.store)
+
     this.store.watch(
       (state, getters) => getters['me/isTeacher'],
-      this.onIsTeacherChanged.bind(this)
+      () => this.onIsTeacherChanged()
     )
 
     if (this.store.getters['me/isTeacher']) {
-      loadSegment()
+      this.onIsTeacherChanged(true)
       window.analytics.ready(this.onInitializeSuccess)
     } else {
       this.onInitializeSuccess()
     }
   }
 
-  async trackPageView (includeIntegrations = {}) {
+  async trackPageView (includeIntegrations = []) {
     await this.initializationComplete
 
-    if (!this.enabled) {
+    if (!this.enabled || this.disableAllTracking) {
       return
     }
 
+    const options = { ...DEFAULT_SEGMENT_OPTIONS }
+    this.addIntegrationsToSegmentOptions(options, includeIntegrations)
+
     const url = `/${Backbone.history.getFragment()}`
```

Comment:
This method returns the query params of the url. Is this something we want in the page tracking?

Reply:
Yep I think so, basing this off of the legacy tracker implementation and that what we had https://github.com/codecombat/codecombat/blob/master/app/core/Tracker.coffee#L69.  Thoughts?

Reply:
Gotcha! Maybe `name` might be an alternative variable name since that's what it becomes in the segment call. Then intent may be more explicit but I don't mind!